jas/runner/jasrunner.go

```diff
@@ -151,15 +152,15 @@ func runIacScan(securityParallelRunner *utils.SecurityParallelRunner, scanner *j
 		defer func() {
 			securityParallelRunner.JasScannersWg.Done()
 		}()
-		results, err := iac.RunIacScan(scanner, module, threadId)
-		if err != nil {
-			return fmt.Errorf("%s %s", clientutils.GetLogMsgPrefix(threadId, false), err.Error())
-		}
+		vulnerabilitiesResults, violationsResults, err := iac.RunIacScan(scanner, module, threadId)
 		securityParallelRunner.ResultsMu.Lock()
 		defer securityParallelRunner.ResultsMu.Unlock()
-		extendedScanResults.IacScanResults = append(extendedScanResults.IacScanResults, results...)
-		err = dumpSarifRunToFileIfNeeded(results, scansOutputDir, jasutils.IaC)
-		return
+		// We first add the scan results and than check for errors to store the exit code to report it in the end
```

Comment:
```suggestion
		// We first add the scan results and only then check for errors, so we can store the exit code in order to report it in the end
```
Fixed typo and rephrased to better understand